atoma-proxy/src/server/handlers/chat_completions.rs

```diff
@@ -765,6 +767,62 @@ async fn handle_streaming_response(
 
     let stream = response.bytes_stream();
 
+    let (event_sender, event_receiver) = flume::unbounded();
+    let (kill_signal_sender, mut kill_signal_receiver) = mpsc::channel(1);
+    let state_manager_sender = state.state_manager_sender.clone();
+    let node_address_clone = node_address.clone();
+    let request_id_clone = request_id.clone();
+    let client_clone = client.clone();
+    tokio::spawn(async move {
+        let mut streamer = Streamer::new(
+            stream,
+            state_manager_sender,
+            selected_stack_small_id,
+            num_input_tokens.unwrap_or(0),
+            estimated_total_tokens,
+            start,
+            node_id,
+            model_name,
+            endpoint,
+        );
+        loop {
+            tokio::select! {
+                chunk = streamer.next() => {
+                    match chunk {
+                        Some(Ok(chunk)) => {
+                            event_sender.send(chunk).unwrap();
+                        }
+                        Some(Err(e)) => {
+                            tracing::error!(target = "atoma-service-streamer", level = "error", "Error sending chunk: {}", e);
+                            continue;
+                        }
+                        None => {
+                            continue;
+                        }
+                    }
+                }
+                _ = kill_signal_receiver.recv() => {
+                    tracing::info!(target = "atoma-service-streamer", "Received kill signal, stopping streamer");
+                    let stop_response = client_clone
+                        .post(format!("{node_address_clone}/v1/stop_generation"))
```

Comment:
in the other PR this was `/v1/stop-streamer`

Reply:
Great observation !